### PR TITLE
Make the proxy jump resolver a bit more flexible.

### LINKED ISSUE #318: Support for more than one jump host
### Description

I have a special server running to which the connection is only possible via 2 jump hosts and my configuration for SSH looks like this:

``` bash
Host jump1
  Hostname                    jump1.corp
  Port                        22
  User                        nightman
  LogLevel                    QUIET

Host jump2
  HostName                    jump2.dmz.corp
  Port                        22
  User                        nightman
  ProxyJump                   jump1
  LogLevel                    QUIET

Host host*.dmz.corp
  ProxyJump                   nightman@jump2:22
```

This is working well with the native SSH clients.

It would be nice when your great libary would support such a configuration as well!

 :-)

### Motivation

Enhancements would make the library better usable for jump hosts.

### Alternatives considered

There is no workaround I found so far.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by nightman68**
fix #318.

### CODE REVIEW

File: sshd-core/src/main/java/org/apache/sshd/client/SshClient.java [hunk 1]
```diff
@@ -683,8 +683,11 @@ protected List<HostConfigEntry> parseProxyJumps(String proxyJump, AttributeRepos
             String host = uri.getHost();
             int port = uri.getPort();
             String userInfo = uri.getUserInfo();
-            HostConfigEntry entry = resolveHost(userInfo, host, port, context, null);
-            jumps.add(entry);
+            HostConfigEntry entry = null;
+            do {
+                entry = resolveHost(userInfo, host, port, context, null);
+                jumps.add(entry);
+            } while((host = entry.getProxyJump()) != null);
```
[the remark this answers is not in the document]
Comment (nightman68) [reply sshd-core/src/main/java/org/apache/sshd/client/SshClient.java:690]: 1. My tests were working like I posted. Using the jumps in reverse oder was not working.
2. I have not tested this. I assumed that people will use scheme 1 or 2, a mix - I don't know.

May be this solution is better than nothing?

**Comment by nightman68** [comment]
1. I fixed the proxy order for "inline" definition - my mistake.
2. The code is now supporting your example configuration above.

**Comment by nightman68** [comment]
Sorry, I do not have the time at the moment getting familiar with your test suite etc.

**Comment by tomaswolf** [comment]
Fixed differently via PR #512 (including tests).

**Comment by nightman68** [comment]
In what release the changes will be included? I can't find this information...

**Comment by tomaswolf** [comment]
See https://github.com/apache/mina-sshd/blob/master/docs/changes/2.13.0.md. It does mention the issue. (Use 2.13.1. Publishing to Maven Central of 2.13.0 omitted the source bundles.)

**Pull request closed without merging**